File: SDKs/googleapis.com/dfareporting/v3.5/java/lib/src/main/java/openapisdk/models/shared/TechnologyTargeting.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * TechnologyTargeting
 * Technology Targeting.
**/
public class TechnologyTargeting {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("browsers")
    public Browser[] browsers;
    public TechnologyTargeting withBrowsers(Browser[] browsers) {
        this.browsers = browsers;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("connectionTypes")
    public ConnectionType[] connectionTypes;
    public TechnologyTargeting withConnectionTypes(ConnectionType[] connectionTypes) {
        this.connectionTypes = connectionTypes;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("mobileCarriers")
    public MobileCarrier[] mobileCarriers;
    public TechnologyTargeting withMobileCarriers(MobileCarrier[] mobileCarriers) {
        this.mobileCarriers = mobileCarriers;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("operatingSystemVersions")
    public OperatingSystemVersion[] operatingSystemVersions;
    public TechnologyTargeting withOperatingSystemVersions(OperatingSystemVersion[] operatingSystemVersions) {
        this.operatingSystemVersions = operatingSystemVersions;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("operatingSystems")
    public OperatingSystem[] operatingSystems;
    public TechnologyTargeting withOperatingSystems(OperatingSystem[] operatingSystems) {
        this.operatingSystems = operatingSystems;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("platformTypes")
    public PlatformType[] platformTypes;
    public TechnologyTargeting withPlatformTypes(PlatformType[] platformTypes) {
        this.platformTypes = platformTypes;
        return this;
    }
}

File: SDKs/googleapis.com/dfareporting/v3.3/python/src/sdk/models/shared/technologytargeting.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import browser as shared_browser
from ..shared import connectiontype as shared_connectiontype
from ..shared import mobilecarrier as shared_mobilecarrier
from ..shared import operatingsystemversion as shared_operatingsystemversion
from ..shared import operatingsystem as shared_operatingsystem
from ..shared import platformtype as shared_platformtype


@dataclass_json
@dataclasses.dataclass
class TechnologyTargeting:
    r"""TechnologyTargeting
    Technology Targeting.
    """
    
    browsers: Optional[list[shared_browser.Browser]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('browsers') }})
    connection_types: Optional[list[shared_connectiontype.ConnectionType]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('connectionTypes') }})
    mobile_carriers: Optional[list[shared_mobilecarrier.MobileCarrier]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('mobileCarriers') }})
    operating_system_versions: Optional[list[shared_operatingsystemversion.OperatingSystemVersion]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('operatingSystemVersions') }})
    operating_systems: Optional[list[shared_operatingsystem.OperatingSystem]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('operatingSystems') }})
    platform_types: Optional[list[shared_platformtype.PlatformType]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('platformTypes') }})
    
